### DISCUSSION TITLE
Add more interrupt skills?

### DISCUSSION THREAD

**Discussion by Cheadstina** [Hekili/hekili#988] [Q&A] [answered]
I have my Havoc DPS Demon Hunter. 
I have setup Interrupts to show up seperately and "Disrupt" works fine. It seems like "Fel Eruption" is never recommended to be used even though Disrupt is on cooldown.

I would like to add "Fel Eruption" to Interrupts - is that done by adding it to the list in Havoc... Toggles... "Utilities / Interrupts"

Same with "Chaos Nova"

**Comment by Cheadstina** [comment, marked as answer]
I think I got it working - a bit difficult to test.
Created a copy ("Havoc with more Interrupts") of the default Havoc (Under Priority list).
Under /hekili Havoc.. Core... Priority.. I selected the new "Havoc with more Interrupts".
Under Havoc Toggles... Utility / Interrupts... add "Fel Eruption" and "Chaos Nova".

Back to the new Priority list "Havoc with more Interrupts"... 
Select Action List "default".
Add a new Entry Chaos Nova.. Select action Chaos Nova. Move it up to before af just after Disrupt.
Same with Fel Eruption.

Am I missing something?

**Reply by Hekili** [reply]
Looks like you've covered what's essentially in the FAQ.

https://github.com/Hekili/hekili/wiki/Frequently-Asked-Questions

Yes, adding to the Interrupts toggle will make it go into the Interrupts display when Show Separately is checked.

**Reply by Cheadstina** [reply]
I did read the faq first - guess it has to be even more explicit before I understands it :/

Am I right in assuming that the skills also have to be added to the priority list?

**Reply by Cheadstina** [reply]
To answer my own question, no, these skills Fel Eruption and Chaos Nova has to be added to the Priority list too before they show up under Interrupts.
I have Show Separately enabled.

**Reply by Hekili** [reply]
Yes, they have to be in the priority list.  That's detailed in the FAQ as well (see about step 6, forward).

**Reply by Cheadstina** [reply]
I'm sorry, it wasn't clear to me. But thanks for your help!

**Reply by Hekili** [reply]
No worries.  Did you get it sorted?

**Reply by Cheadstina** [reply]
I can't get it working exactly as Disrupt does.
I've created a new priority list(and selected it for Havoc), added Fel Eruption and Chaos Nova as priority 1 and 2. (Disrupt is third priority).
I have added Fel Eruption and Chaos Nova in Utilities/Interrupts under Toggles.

When I get into combat, Fel Eruption is show in Interrupt - even though the mob isn't casting anything. I guess that's because Fel Eruption is a stun?
Same with Chaos Nova.
Disrupt is only appearing if the targeted mob is casting something.

Interrupts are set to show Interrupts and show separately.

**Reply by Hekili** [reply]
Actual interrupts have a built in check to be timed immediately (for a channeled spell) or in the last second (for any other spell).

The closest approximation you can make in-game would be:

`debuff.casting.up & ( debuff.casting.v3 | debuff.casting.remains < 1 )`

Explanation:
* `debuff.casting.up` = the target is casting
* `debuff.casting.v3` = the spell is channeled
* `debuff.casting.remains < 1` = there is less than 1 second left on the cast.

So that ends up being:
Either the target is casting and the spell is channeled OR the target is casting and there's less than 1 second left on the cast.

**Reply by Cheadstina** [reply]
Thanks, that seems to work very well.
I'm not sure about this one though?
"debuff.casting.remains < 1 = there is less than 1 second left on the cast."
Why should it show up if there's less than 1 second left of the cast? I thought it should only show up if there's more than 1 second?
Or am I misunderstanding it again? :)

**Reply by Hekili** [reply]
If your target is casting, you might as well let them keep casting so they're not doing other things.  That's why you typically interrupt at the end of a cast, rather than the start.

**Reply by Cheadstina** [reply]
That makes sense - I'd just like to have the interrupt icon up when they are casting. Then I'll time the interrupt myself.
As I understand the interrupt icon only shows if there's less than 1 second left of the target's cast?

Then I guess it should just be:
debuff.casting.up & ( debuff.casting.v3 | debuff.casting.remains < 10 )
